guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -715,6 +715,424 @@ You need to read configuration from different sources (like database or filesyst
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+//span[contains(text(), 'SomeTextHere')]
+```
+
+### Not recommended
+
+```xml
+//span[text()='SomeTextHere']
+```
+
+{% endcollapsible %}
+
+11.3.3.8. **Selectors SHOULD be built in the proper order.** When building selectors for form elements, start with the parent context of the form element. Then specify the element `name` attribute in your selector to ensure the correct element is targeted. To build a selector for an input, use the pattern: `{{section_selector}}` `{{input_selector}}` or for a button: `{{section_selector}}` `{{button_selector}}`
+
+**Why?** Traversing the DOM takes a finite amount of time and reducing the scope of the selector makes the selector lookup as efficient as possible.
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+<element name="productName" type="input" selector="*[data-index='product-details'] input[name='product[name]']"/>
+```
+
+### Not recommended
+
+```xml
+<element name="productName" type="input" selector=".admin__field[data-index=name] input"/>
+```
+
+{% endcollapsible %}
+
+11.3.3.9. **Selectors SHOULD be built with appropriate specificity.** Selectors that are too general might sweep up unexpected elements. When possible, select the first parent tag and then specify the desired element within that selection.
+
+**Why?** Elements that are overly specific are less flexible and may fail if unexpected DOM changes occur. It also reduces the amount of the DOM it needs to parse.
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+form[name='myform'] > input[name='firstname']
+
+//*[@id='container'][@class='dashboard-title']
+```
+
+### Not recommended
+
+```xml
+input[name='firstname']
+
+//*[@id='container']/*[@class='dashboard-advanced-reports']/*[@class='dashboard-advanced- reports-description']/*[@class='dashboard-title']
+```
+
+{% endcollapsible %}
+
+11.3.3.10. Parameterized selectors MUST use descriptive variable names.
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+<element name="storeName" type="checkbox" selector="//label[contains(text(),'{{storeName}}')]" parameterized="true"/>
+```
+
+### Not recommended
+
+```xml
+<element name="storeName" type="checkbox" selector="//label[contains(text(),'{{var1}}')]" parameterized="true"/>
+```
+
+{% endcollapsible %}
+
+#### 11.3.4. Data Entities
+
+11.3.4.1. Data entities SHOULD make use of `unique="suffix"` or `unique="prefix"` to ensure that tests using the entity can be repeatedely ran against the same environment.
+
+11.3.4.2. Data entities MUST NOT be modified and MUST NOT merge additional fields without complete understanding and verifying the usage of existing data in tests. New data entities SHOULD be created if you are not sure.
+
+11.3.4.3. Data entity file names MUST follow this pattern:
```

Comment:
This is minor. How about move 11.3.4.3 as 11.3.4.1? so that we always talk about file name pattern first if applicable.

Reply:
Fixed